src/hooks/frapi/useSearchRescueApi.js

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useReducer, useState } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { isValidUuidV4 } from '~/helpers/uuidValidator'
+import { getRescue } from '~/store/actions/rescues'
+
+export default function useSearchRescueApi (initialRescueId = null) {
+  const [rescueId, setRescueId] = useState(initialRescueId)
+  const reduxDispatch = useDispatch()
+
+  const rescueFetchReducer = (state, action) => {
```

Comment:
This reducer could be defined outside of the function just so it doesn't have to get defined every call.